### ISSUE TITLE
Bug: 1.12 makes brotli non-optional

### ISSUE THREAD

**Issue by peterschutt** [starlite-api/starlite#390]
**Describe the bug**
hello-world app won't start on `1.12`

**To Reproduce**
Try to run an app without brotli extras

**Additional context**
```python-traceback
Traceback (most recent call last):
  File "/home/peter/.cache/pypoetry/virtualenvs/starlite-hello-world-nPMVAxaQ-py3.10/bin/uvicorn", line 8, in <module>
    sys.exit(main())
  File "/home/peter/.cache/pypoetry/virtualenvs/starlite-hello-world-nPMVAxaQ-py3.10/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/peter/.cache/pypoetry/virtualenvs/starlite-hello-world-nPMVAxaQ-py3.10/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/peter/.cache/pypoetry/virtualenvs/starlite-hello-world-nPMVAxaQ-py3.10/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/peter/.cache/pypoetry/virtualenvs/starlite-hello-world-nPMVAxaQ-py3.10/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/peter/.cache/pypoetry/virtualenvs/starlite-hello-world-nPMVAxaQ-py3.10/lib/python3.10/site-packages/uvicorn/main.py", line 407, in main
    run(
  File "/home/peter/.cache/pypoetry/virtualenvs/starlite-hello-world-nPMVAxaQ-py3.10/lib/python3.10/site-packages/uvicorn/main.py", line 575, in run
    server.run()
  File "/home/peter/.cache/pypoetry/virtualenvs/starlite-hello-world-nPMVAxaQ-py3.10/lib/python3.10/site-packages/uvicorn/server.py", line 60, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "/home/peter/.pyenv/versions/3.10.5/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/home/peter/.pyenv/versions/3.10.5/lib/python3.10/asyncio/base_events.py", line 646, in run_until_complete
    return future.result()
  File "/home/peter/.cache/pypoetry/virtualenvs/starlite-hello-world-nPMVAxaQ-py3.10/lib/python3.10/site-packages/uvicorn/server.py", line 67, in serve
    config.load()
  File "/home/peter/.cache/pypoetry/virtualenvs/starlite-hello-world-nPMVAxaQ-py3.10/lib/python3.10/site-packages/uvicorn/config.py", line 479, in load
    self.loaded_app = import_from_string(self.app)
  File "/home/peter/.cache/pypoetry/virtualenvs/starlite-hello-world-nPMVAxaQ-py3.10/lib/python3.10/site-packages/uvicorn/importer.py", line 24, in import_from_string
    raise exc from None
  File "/home/peter/.cache/pypoetry/virtualenvs/starlite-hello-world-nPMVAxaQ-py3.10/lib/python3.10/site-packages/uvicorn/importer.py", line 21, in import_from_string
    module = importlib.import_module(module_str)
  File "/home/peter/.pyenv/versions/3.10.5/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/peter/PycharmProjects/starlite-hello-world/./main.py", line 6, in <module>
    from starlite import Starlite, get
  File "/home/peter/.cache/pypoetry/virtualenvs/starlite-hello-world-nPMVAxaQ-py3.10/lib/python3.10/site-packages/starlite/__init__.py", line 14, in <module>
    from .app import Starlite
  File "/home/peter/.cache/pypoetry/virtualenvs/starlite-hello-world-nPMVAxaQ-py3.10/lib/python3.10/site-packages/starlite/app.py", line 10, in <module>
    from starlite.config import (
  File "/home/peter/.cache/pypoetry/virtualenvs/starlite-hello-world-nPMVAxaQ-py3.10/lib/python3.10/site-packages/starlite/config/__init__.py", line 2, in <module>
    from .compression import CompressionConfig
  File "/home/peter/.cache/pypoetry/virtualenvs/starlite-hello-world-nPMVAxaQ-py3.10/lib/python3.10/site-packages/starlite/config/compression.py", line 7, in <module>
    from starlite.middleware.compression.brotli import BrotliMode
  File "/home/peter/.cache/pypoetry/virtualenvs/starlite-hello-world-nPMVAxaQ-py3.10/lib/python3.10/site-packages/starlite/middleware/compression/brotli.py", line 18, in <module>
    raise MissingDependencyException("brotli is not installed") from e
starlite.exceptions.exceptions.MissingDependencyException: brotli is not installed
```

**Comment by peterschutt** [comment]
From 53d3a24d709941afce906fe2333459076f312fde - where the `BrotliMode` enum was moved from `enums.py` into `starlite.middleware.compression.brotli`, and then accessed from within `starlite.config.compression` causing the execution path through the brotli missing dependency check.

**Comment by Goldziher** [comment]
Can't we simply eliminate the enum completely?

**Comment by peterschutt** [comment]
Yep, I put a PR through that just moves the enums into compression config - I think they are more related to the config than the middleware implementation anyway, but I'm not fussed whether we roll with the enums, or just use a `Literal`.

**Comment by peterschutt** [comment]
closed in #391
